src/nplinker/metabolomics/spectrum.py

```diff
@@ -1,214 +1,88 @@
 from __future__ import annotations
+from functools import cached_property
 from typing import TYPE_CHECKING
+import numpy as np
 from nplinker.strain import Strain
 from nplinker.strain_collection import StrainCollection
-from nplinker.utils import sqrt_normalise
 
 
 if TYPE_CHECKING:
     from .molecular_family import MolecularFamily
 
-GNPS_KEY = "gnps"
-
-JCAMP = (
-    "##TITLE={}\\n"
-    + "##JCAMP-DX=nplinker vTODO\\n"
-    + "##DATA TYPE=Spectrum\\n"
-    + "##DATA CLASS=PEAKTABLE\\n"
-    + "##ORIGIN=TODO_DATASET_ID\\n"
-    + "##OWNER=nobody\\n"
-    + "##XUNITS=M/Z\\n"
-    + "##YUNITS=RELATIVE ABUNDANCE\\n"
-    + "##NPOINTS={}\\n"
-    + "##PEAK TABLE=(XY..XY)\\n"
-    + "{}\\n"
-    + "##END=\\n"
-)
-
 
 class Spectrum:
-    def __init__(self, id, peaks, spectrum_id: str, precursor_mz, parent_mz=None, rt=None):
-        self.id = id
-        self.peaks = sorted(peaks, key=lambda x: x[0])  # ensure sorted by mz
-        self.normalised_peaks = sqrt_normalise(self.peaks)  # useful later
-        self.n_peaks = len(self.peaks)
-        self.max_ms2_intensity = max(intensity for mz, intensity in self.peaks)
-        self.total_ms2_intensity = sum(intensity for mz, intensity in self.peaks)
-        self.spectrum_id = spectrum_id  # MS1.name
-        self.rt = rt
-        # TODO CG: should include precursor mass and charge to calculate precursor_mz
-        # parent_mz can be calculate from precursor_mass and charge mass
+    def __init__(
+        self,
+        spectrum_id: str,
+        mz: list[float],
+        intensity: list[float],
+        precursor_mz: float,
+        rt: float = 0,
+        metadata: dict | None = None,
+    ) -> None:
+        """Class to model MS/MS Spectrum.
+
+        Args:
```

Comment:
Same comment as in a previous PR, I find using both args and attributes in classes docstrings a bit redundant and not necessary. But it's up to you :)

Reply:
Clarity is over redundancy in this case ;-)